File: bestan-lua/src/test/java/bestan/common/lua/TestLua.java
package bestan.common.lua;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@LuaAnnotation(optional=true)
public class TestLua implements ILuaConfig {
	public int test = 11;
	public List<Integer> nums = new ArrayList<Integer>();
	public Map<Integer, ValueLua> map = new HashMap<>();
	public List<ValueLua> lists = new ArrayList<>();
	public String string = "aaa";
	public Map<Integer, Map<Integer, Integer>> dmap = new HashMap<>();
	public List<Map<Integer, Map<Integer, Map<Integer, Integer>>>> dList = new ArrayList<>();
	@LuaParamAnnotation(optional=true)
	public long lvalue = 122313L;
	
	public static class ValueLua implements ILuaConfig{
		public Integer v1 = 0;
		public Integer v2 = 0;
	}
}

File: bestan-lua/TestLua.lua
local map = {
    test = 1111,
    nums = { 1, 2, 3 },
    map = {
        [1] = { v1 = 11, v2 = 100,  },
        [20] = { v1 = 100, v2 = 1000, },
    },
    lists = {
        {v1 = 11, v2 = 22},
        {v1 = 111, v2 = 222},
    },
    string = "123123",
    dmap = {
        [1] = {
            [10] = 11,
        },
    },
    dList = {
        { [10]={
            [1] = {
                [10] = 11,
                },
            },
        },
    },
    number = "aaa",
    --lvalue = "123123123123123",
}
print("aaaa")

return map;